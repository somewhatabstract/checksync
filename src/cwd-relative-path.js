// @flow
import path from "path";

export default function(filePath: string) {
    return path.relative(process.cwd(), filePath);
}
