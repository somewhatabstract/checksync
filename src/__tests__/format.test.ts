import chalk from "chalk";
import * as CwdRelativePath from "../cwd-relative-path";

import Format from "../format";

describe("Format", () => {
    describe("#verbose", () => {
        it("should prefix with Verbose", () => {
            // Arrange

            // Act
            const result = Format.verbose("Test");

            // Assert
            expect(result).toMatchSnapshot();
        });

        it("should render with chalk colors", () => {
            // Arrange
            chalk.level = 1;

            // Act
            const result = Format.verbose("Test");

            // Assert
            expect(result).toMatchSnapshot();
        });
    });

    describe("#info", () => {
        it("should prefix with _INFO__", () => {
            // Arrange

            // Act
            const result = Format.info("Test");

            // Assert
            expect(result).toMatchSnapshot();
        });

        it("should render with chalk colors", () => {
            // Arrange
            chalk.level = 1;

            // Act
            const result = Format.info("Test");

            // Assert
            expect(result).toMatchSnapshot();
        });
    });

    describe("#error", () => {
        it("should prefix with _ERROR__", () => {
            // Arrange

            // Act
            const result = Format.error("Test");

            // Assert
            expect(result).toMatchSnapshot();
        });

        it("should render with chalk colors", () => {
            // Arrange
            chalk.level = 1;

            // Act
            const result = Format.error("Test");

            // Assert
            expect(result).toMatchSnapshot();
        });
    });

    describe("#warn", () => {
        it("should prefix with _WARNING__", () => {
            // Arrange

            // Act
            const result = Format.warn("Test");

            // Assert
            expect(result).toMatchSnapshot();
        });

        it("should render with chalk colors", () => {
            // Arrange
            chalk.level = 1;

            // Act
            const result = Format.warn("Test");

            // Assert
            expect(result).toMatchSnapshot();
        });
    });

    describe("#mismatch", () => {
        it("should prefix with _MISMATCH__", () => {
            // Arrange

            // Act
            const result = Format.mismatch("Test");

            // Assert
            expect(result).toMatchSnapshot();
        });

        it("should render with chalk colors", () => {
            // Arrange
            chalk.level = 1;

            // Act
            const result = Format.mismatch("Test");

            // Assert
            expect(result).toMatchSnapshot();
        });
    });

    describe("#fix", () => {
        it("should prefix with _FIX_", () => {
            // Arrange

            // Act
            const result = Format.fix("Test");

            // Assert
            expect(result).toMatchSnapshot();
        });

        it("should render with chalk colors", () => {
            // Arrange
            chalk.level = 1;

            // Act
            const result = Format.fix("Test");

            // Assert
            expect(result).toMatchSnapshot();
        });
    });

    describe("#cwdFilePath", () => {
        it("should render with chalk colors", () => {
            // Arrange
            chalk.level = 1;

            // Act
            const result = Format.cwdFilePath("Test");

            // Assert
            expect(result).toMatchSnapshot();
        });

        it("should turn path into relative path", () => {
            // Arrange
            jest.spyOn(CwdRelativePath, "default").mockReturnValue(
                "RELATIVE_PATH",
            );

            // Act
            const result = Format.cwdFilePath("/Users/test/test/test.js");

            // Assert
            expect(result).toBe("RELATIVE_PATH");
        });
    });

    describe("#code", () => {
        it("should render with chalk colors", () => {
            // Arrange
            chalk.level = 1;

            // Act
            const result = Format.code("Test");

            // Assert
            expect(result).toMatchSnapshot();
        });
    });

    describe("#heading", () => {
        it("should render with chalk colors", () => {
            // Arrange
            chalk.level = 1;

            // Act
            const result = Format.heading("Test");

            // Assert
            expect(result).toMatchSnapshot();
        });
    });
});
