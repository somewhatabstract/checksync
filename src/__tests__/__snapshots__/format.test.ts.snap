// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Format #code should render with chalk colors 1`] = `"[1m[33mTest[39m[22m"`;

exports[`Format #cwdFilePath should render with chalk colors 1`] = `"[90mTest[39m"`;

exports[`Format #error should prefix with _ERROR__ 1`] = `"Error    Test"`;

exports[`Format #error should render with chalk colors 1`] = `"[31mError   [39m Test"`;

exports[`Format #fix should prefix with _FIX_ 1`] = `"Fix      Test"`;

exports[`Format #fix should render with chalk colors 1`] = `"[1m[92mFix     [39m[22m Test"`;

exports[`Format #heading should render with chalk colors 1`] = `"[1m[32mTest[39m[22m"`;

exports[`Format #info should prefix with _INFO__ 1`] = `"Info     Test"`;

exports[`Format #info should render with chalk colors 1`] = `"[34mInfo    [39m Test"`;

exports[`Format #mismatch should prefix with _MISMATCH__ 1`] = `"Mismatch Test"`;

exports[`Format #mismatch should render with chalk colors 1`] = `"[1m[93mMismatch[39m[22m Test"`;

exports[`Format #verbose should prefix with Verbose 1`] = `"Verbose  Test"`;

exports[`Format #verbose should render with chalk colors 1`] = `"[90mVerbose [39m [2mTest[22m"`;

exports[`Format #warn should prefix with _WARNING__ 1`] = `"Warning  Test"`;

exports[`Format #warn should render with chalk colors 1`] = `"[33mWarning [39m Test"`;
